Automattic/wp-calypso
# State: Move post notices away from middleware

#### Changes proposed in this Pull Request

This PR moves the post notices away from the notices middleware and into the corresponding action thunks. This way these notices get modularized together with the posts state module and don't clutter the main entry point.

We also refactor some tests to use `jest` because that makes it easier to test nested thunks.

#### Testing instructions

* Try saving, deleting, and restoring to a post revision and verify notices still appear as they do in production.
* Verify all tests still pass.
